### PR TITLE
Llama2 70b

### PR DESCRIPTION
**Pull request by dan-garvey**
This is primarily @PhaneeshB's work, just prepping it to merge into main

### CODE REVIEW

**Comment by dan-garvey** [comment]
@PhaneeshB @Abhishek-Varma I haven't been able to test all the variations on this due to some weird torch-mlir errors after newest brevitas branch. I think its right in theory but I'd rather it worked before landing it

File: apps/language_models/scripts/vicuna.py [hunk 1]
```diff
@@ -855,7 +854,7 @@ def compile_to_vmfb_one_model(
                     print(f"[DEBUG] converting torch to linalg")
                     run_pipeline_with_repro_report(
                         module0,
-                        "builtin.module(func.func(torch-unpack-torch-tensor),torch-backend-to-linalg-on-tensors-backend-pipeline)",
+                        "builtin.module(func.func(torch-unpack-quant-tensor),func.func(torch-convert-custom-quant-op),torch-backend-to-linalg-on-tensors-backend-pipeline)",
```
Comment (jinchen62) [inline apps/language_models/scripts/vicuna.py:857]: Could you rebase?

File: apps/language_models/scripts/vicuna.py [hunk 2]
```diff
@@ -1456,9 +1455,10 @@ def compile(self, download_vmfb=False):
                 else:
                     compilation_prompt = "".join(["0" for _ in range(17)])
 
-                if Path(f"first_{self.precision}.mlir").exists():
-                    print(f"loading first_{self.precision}.mlir")
-                    with open(Path(f"first_{self.precision}.mlir"), "r") as f:
+                fllama_path = f"first_{self.model_name}_{self.precision}.mlir"
```
Comment (jinchen62) [inline apps/language_models/scripts/vicuna.py:1458]: Could you rename it like first_model_path, cuz the model name might differ and "first" would be more clear.

File: apps/language_models/scripts/vicuna.py [hunk 3]
```diff
@@ -1533,13 +1533,14 @@ def compile(self, download_vmfb=False):
...
-                        with open(f"first_{self.precision}.mlir", "w+") as f:
+                        with open(fllama_path, "w+") as f:
                             f.write(first_module)
                         print("Finished writing IR after dynamic")
-
-                if Path(f"second_{self.precision}.mlir").exists():
-                    print(f"loading second_{self.precision}.mlir")
-                    with open(Path(f"second_{self.precision}.mlir"), "r") as f:
+                print(f"[DEBUG] Starting generation of second llama")
+                sllama = f"second_{self.model_name}_{self.precision}.mlir"
```
Comment (jinchen62) [inline apps/language_models/scripts/vicuna.py:1540]: Same comment as fllama.

File: apps/language_models/scripts/vicuna.py [hunk 4]
```diff
@@ -1836,7 +1845,7 @@ def create_prompt(model_name, history):
             precision=args.precision,
             vicuna_mlir_path=vic_mlir_path,
             vicuna_vmfb_path=vic_vmfb_path,
-            load_mlir_from_shark_tank=args.load_mlir_from_shark_tank,
+            load_mlir_from_shark_tank=False,
```
Comment (jinchen62) [inline apps/language_models/scripts/vicuna.py:1848]: Don't we want to control it by flag?

File: apps/language_models/src/model_wrappers/vicuna_model.py [hunk 5]
```diff
@@ -580,25 +582,532 @@ def forward(
         return tuple(return_vals)
 
 
-class CombinedModel(torch.nn.Module):
```
Comment (jinchen62) [inline apps/language_models/src/model_wrappers/vicuna_model.py:583]: We don't need CombinedModel anymore?

File: apps/language_models/src/model_wrappers/vicuna_model.py [hunk 6]
```diff
@@ -580,25 +582,532 @@ def forward(
...
+        i154,
+        i155,
+        i156,
+        i157,
+        i158,
+        i159,
+        i160,
+    ):
+        # input_ids = input_tuple[0]
+        # input_tuple = torch.unbind(pkv, dim=0)
```
Comment (jinchen62) [inline apps/language_models/src/model_wrappers/vicuna_model.py:783]: Clean it up?

**Review by jinchen62 — APPROVED** [review]

**Pull request merged**